### ISSUE TITLE
HTTP Metrics & Tracing: provide a way to Opt-In `url.scheme`

### ISSUE THREAD

**Issue by antonfirsov** [dotnet/runtime#104738]
According to the spec, `url.scheme` should be Opt-In:
https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#metric-httpclientrequestduration
https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#http-client

Given that currently we don't have a mechanism for selectively opting into attributes + for historic reasons, we are reporting `url.scheme` unconditionally in both Metrics and Tracing. To deal with this on per-client basis, we need an API.

Adding this to 10.0 for now, but we can triage out to Future if it's not critical.

**Comment by julealgon** [comment]
@antonfirsov do you know why it was decided that this attribute would be opt-in? Is it due to noise/payload size considerations vs usefulness of the information, or is there more to it?

**Comment by antonfirsov** [comment]
I defer to @lmolkova to answer that if she's around :)

**Comment by JamesNK** [comment]
It's weird that it's opt-in on the client but required on the server.

I don't know the reason. Perhaps it's related to redirects? i.e. a redirect from http address to https

**Comment by lmolkova** [comment]
you have scheme as a part of `url.full` on the client, but url is not populated on the server. Opt-in in this case means something around the lines of "it seems it was useful in some edge cases and just redundant/noisy in others".

I.e. it's not considered that useful. LMK if you believe it is useful.

**Comment by antonfirsov** [comment]
@lmolkova there is no `url.full` in the duration metric only in the span. Are we sure we aren't removing an important piece of info? I don't see other attributes that would help determining if was a secure request.

**Comment by lmolkova** [comment]
Ah, I didn't realize the question is for metrics. The discussion on `url.scheme` for client metrics - https://github.com/open-telemetry/semantic-conventions/issues/450

TL;DR: scheme and port have 1:1 relationship, since we have the port, users can figure out scheme from other telemetry or by knowing their setup.  I.e. it's not essential. Still, for some users may prefer to have scheme recorded when it's important and to save an extra step when analyzing things.

I believe there are some reasons (not super-strong) in otel to have the same requirement level on traces and metrics, so it's opt-in to reduce traces volume. I imagine how url.scheme can be changed to recommended if there is a strong push/feedback for it.

For .NET if you believe having scheme on client metrics is important, I'd rather populate it by default instead of providing opt-in. The extra attribute on metric is much more tolerable than on spans, but having opt-in API is a burden.

**Comment by antonfirsov** [comment]
Today we can only implement this with a switch/environment variable `System.Net.Http` which feels odd and might be outdated in the future by a generic mechanism for controlling built-in attributes. We should solve this if/when such a mechanism will be available.

In the meanwhile, we can decide to completely remove `url.scheme` to reduce trace volume, but IMO we should do it only if there will be customer complaints.
